wazimap_ng/datasets/hooks.py

```diff
@@ -3,152 +3,51 @@
 
 def process_task_info(task):
     """
     Process task
     """
     notify = task.kwargs.get("notify", False)
-    assign_task = task.kwargs.get("assign", False)
-    obj = next(iter(task.args))
-
-    if assign_task:
-        obj.task = task
-        obj.save()
 
     if notify:
         notification_type = "success" if task.success else "error"
-
-        session_key = task.kwargs.get("key", False)
+        msg = "Successfully" if task.success else "Failed to"
         task_type = task.kwargs.get("type", False)
-        message = task.kwargs.get("message", None)
-        results = task.result or {}
-
-        # Get message
-        notify = Notify()
-        if not message:
-            message = notify.get_nofitification_details(notification_type, obj, task_type, results)
-
-        # Add message to user
-        notify_user(notification_type, session_key, message, task.id)
-
-def notify_user(notification_type, session_key, message, task_id=None):
-    """
-    Call back function after the task has been executed.
-    This function gets passed the complete Task object as its argument.
-
-    We get session key from kwargs of the task object and data of created object
-    is passes through results value in task object.
-    """
-    session = Session.objects.filter(session_key=session_key).first()
-
-    if session:
-        decoded_session = custom_admin_notification(
-            session.get_decoded(), notification_type, message, task_id
-        )
-        session.session_data = Session.objects.encode(decoded_session)
-        session.save()
-
-def custom_admin_notification(session, notification_type, message, task_id=None):
-    """
-    Function for implementing custom admin notifications.
-    notifications are stored in session and show to user when user refreshes page.
-
-    A valid session object must be passed to this function with notification type and message
-    
-    Type of notifications:
-        * success
-        * info
-        * warning
-        * error
-
-    message is able to parse html so we can use html tags and classes while creating message
-    """
-    notification = {"type": notification_type, "message": message}
-    messages = []
-
-    if task_id:
-        notification["task_id"] = task_id
-
-    if "notifications" in session:
-        messages = session['notifications']
-    messages.append(notification)
-    session['notifications'] = messages
-    return session
-
-def add_to_task_list(session, task):
-    """
-    Add task id to session task_list
-    """
-    task_list = session.get("task_list", [])
-    task_list.append(task)
-    session["task_list"] = task_list
+        notification = Notification.objects.filter(task_id=task.id).first()
+
+        if notification:
+            notification.level = notification_type
+            verb = get_notification_verb(task_type, notification_type)
+            if verb:
+                notification.verb = verb
+            notification.unread = True
```

Comment:
if I read this correctly instead of generating a new notification you update the existing one to be read? 

Why not add a new one? What is the rationale?

Reply:
Requirement for the task was to show task in progress.

So to complete that what we do is:
* Start a notification and mark it in progress.
* In progress tasks are showed in separate area in notifications.
* When Django-Q is done processing that task fetch notification linked to that specific task and change language of the task that is verb and set it as unread.
* Change level of task to the required on: error, warning or success and it will be removed from in progress and shifted to other notification.
* Where a dot will appear so user will know this is new notification for task completed